server/backends/chunkstore/chunkstore.go

```diff
@@ -462,65 +470,103 @@ func (l *writeLoop) write() error {
 	return err
 }
 
-func (l *writeLoop) readFromWriteChannel() {
+func (l *writeLoop) readFromWriteChannel() *WriteRequest {
 	l.timeout = false
-	var p []byte
 	select {
-	case p, l.open = <-l.writeChannel:
-		if l.open && p == nil {
-			l.flushTime = time.Unix(0, 0)
-		} else {
-			l.chunk = append(l.chunk, p...)
-			l.lastWriteSize = len(p)
+	case req, open := <-l.writeChannel:
```

Comment:
nit: consider having `writeChannel` be a stream of `*WriteRequest` rather than `WriteRequest` since you return it by reference here anyway